### PR TITLE
Addison Cole - C10 Edges

### PR DESCRIPTION
**Pull request by addisoncole**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | Pretty much all of it. Objects are pretty much hashes, arrays are still arrays, I just have fewer of the built in enumerables and functions that create a simpler code in ruby, so sometimes i had to write more lines for the same thing in javascript
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | Not really, I like the Ruby docs a bit more, they seem a little more accessible, but I typically searched for a concept i was trying to replicate from ruby, if I couldnt find what i was looking for in the docs, or if i was confused about it, i would use stack exchange for more understanding or similar examples, or different methods of doing the same thing. 
What was a challenge you were able to overcome on this assignment? | I didn't find this too challenging, I think in some ways I may have improved the code from my ruby version of this, and in other ways it had to be elongated because JS was missing some of the more powerful enumerables. I tried to keep things dry by creating additional methods. I wouldve liked to explore how to make these helper methods that were not quite part of the assignment as private methods, but decided to put time elsewhere.  I think the hardest challenge was getting the logic right at the end in the highestScoreFrom action.
What is your favorite thing about learning a new programming language? | learning a new language! figuring out why you might choose this application of a language over another!
What is your least favorite thing about learning a new programming language? | Feeling like I am starting over in knowledge and not knowing as many shortcuts and things that make programming in that language easier or more accessible or quicker
Do you have any recommendations on how we could improve this project for the next cohort? | thought it was a great practice! I felt much better about JS after working on it.

### CODE REVIEW

**Comment by tildeee** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | x
Small commits with meaningful commit messages | x
**Code Requirements** |
`drawLetters` method | x
Uses appropriate data structure to store the letter distribution | x
All tests for `drawLetters` pass | x
`usesAvailableLetters` method | x
All tests for `usesAvailableLetters` pass | x
`scoreWord` method | x
Uses appropriate data structure to store the letter scores | x
All tests for `scoreWord` pass | x
`highestScoreFrom` method | x
Appropriately handles edge cases for tie-breaking logic | x
All tests for `highestScoreFrom` pass | x
**Overall** |

Fantastic code! Your code looks great, and I reeeeally like your solution for `usesAvailableLetters`, it's my favorite solution! I have a few small comments about syntax, but no worries-- Your code overall looks good, good work on this project.

File: src/adagrams.js [hunk 1]
```diff
@@ -1,7 +1,139 @@
...
+      return 4
+      case 'K':
+      return 5
+      case 'J':
+      case 'X':
+      return 8
+      case 'Q':
+      case 'Z':
+      return 10
+    }
```
Comment (tildeee) [inline src/adagrams.js:60]: Good approach using a switch/case conditional! You probably could refactor this to use an object (a hash) though. I like that this is a separate function by the way

File: src/adagrams.js [hunk 2]
```diff
@@ -1,7 +1,139 @@
...
+  },
+
   drawLetters() {
-    // Implement this method for wave 1
+
+    let letterPool = Adagrams.createLetterPool()
+    let playerHand = []
```
Comment (tildeee) [inline src/adagrams.js:66]: missing semicolons

File: src/adagrams.js [hunk 3]
```diff
@@ -1,7 +1,139 @@
...
+
+    for (let i = 0; i < 10; i += 1) {
+      let randomNum = Math.floor ( Math.random() * letterPool.length )
```
Comment (tildeee) [inline src/adagrams.js:69]: nit picking: you probably don't want a space after a function call (aka `Math.floor ( Math.random ...)` -> `Math.floor( Math.random ...)` )

File: src/adagrams.js [hunk 4]
```diff
@@ -1,7 +1,139 @@
...
+      let letter = letterPool[randomNum];
+      letterPool.splice(randomNum, 1);
+      playerHand.push(letter);
+    }
+
+    return playerHand
```
Comment (tildeee) [inline src/adagrams.js:75]: semicolon!

File: src/adagrams.js [hunk 5]
```diff
@@ -1,7 +1,139 @@
...
+    let i = 0;
+    while (i < checkWord.length) {
+      if (checkWord[i] === checkLetters[i]) {
+        i++;
+      } else {
+        return false
+      }
+    }
+
+    return true
```
Comment (tildeee) [inline src/adagrams.js:98]: semicolons!

Comment (tildeee) [inline src/adagrams.js:98]: I love this approach! I think this is a really great approach to sort them first :) Makes it more efficient.

File: src/adagrams.js [hunk 6]
```diff
@@ -1,7 +1,139 @@
...
+        } else if (word.length === 10) {
+          highestScoringWord.word = word;
+          highestScoringWord.score = Adagrams.scoreWord(word);
+        } else if (word.length < highestScoringWord.word.length) {
+          highestScoringWord.word = word;
+          highestScoringWord.score = Adagrams.scoreWord(word);
+        }
+      }
+    });
+    return highestScoringWord
```
Comment (tildeee) [inline src/adagrams.js:134]: Nicely done. There's a few missing semicolons. That being said, great work
